src/builtins.ts

```diff
@@ -134,9 +133,13 @@ export function compileCall(
       compiler.currentType = Type.bool;
       if (!type) return module.createUnreachable();
       let classType = type.classReference;
-      return classType !== null && classType.lookupOverload(OperatorKind.INDEXED_GET) !== null
-        ? module.createI32(1)
-        : module.createI32(0);
+      return (
+        classType !== null &&
+        classType.lookupField("buffer_") !== null &&
+        classType.lookupField("length_") !== null &&
+        classType.lookupOverload(OperatorKind.INDEXED_GET) !== null &&
+        classType.lookupOverload(OperatorKind.INDEXED_SET) !== null
+      ) ? module.createI32(1) : module.createI32(0);
```

Comment:
Can we leave this for a later PR? Otherwise, it should check whether `Array<*>` is in the prototype chain.

Reply:
Sure

Reply:
Done!

Reply:
To bubble up all levels, for example in an `A extends B extends C extends Array<T>` case:

```ts
var extendsArray = false;
var current = classType;
do {
  if (current.prototype == program.arrayPrototype) {
    extendsArray = true;
    break;
  }
} while (current = current.base);
```

Reply:
I think make sense create function for that and called it like `hasPrototypeOf(proto1, proto2)`. WDYT?

Reply:
I found this: https://github.com/AssemblyScript/assemblyscript/blob/2131c519321ecfb0b1c1a642ba9d36f3c18a81cc/src/program.ts#L2937

Reply:
Sure, makes sense. Maybe `ClassPrototype#extends`, so one can do `if (classType.prototype.extends(program.arrayPrototype)) { ... }`?

Reply:
Yeah, though `isAssignableTo` does this on concrete types (resolved generic arguments), not just prototypes (unresolved generic arguments). That's the check when assigning a reference to a reference var.